source/common/http/http1/codec_impl.h

```diff
@@ -295,8 +314,8 @@ class ClientConnectionImpl : public ClientConnection, public ConnectionImpl {
 
   // Http::ClientConnection
   StreamEncoder& newStream(StreamDecoder& response_decoder) override;
-  void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {}
-  void onUnderlyingConnectionBelowWriteBufferLowWatermark() override {}
+  void onUnderlyingConnectionAboveWriteBufferHighWatermark() override;
```

Comment:
One thought here from looking through this PR. Would it be easier to understand if there was symmetry for the onUnderlying* calls on the server connection also? (So basically move them into the base connection interface). I think currently connection manager just deals with invoking callbacks directly. Just wondering if codec should actually do it in both cases...